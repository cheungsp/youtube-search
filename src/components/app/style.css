body {
  background-color: whitesmoke;
}

.video-item img {
  max-width: 64px;
}

.video-detail .details {
  margin-top: 10px;
  padding: 10px;
  border: 1px solid #ddd;
  border-radius: 4px;
}