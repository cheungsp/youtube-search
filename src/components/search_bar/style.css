.search-bar {
  margin: 20px;
  text-align: center;
}

.search-bar input {
  width: 80%;
}

/*
"test": "jest"
"test": "react-scripts test --env=jsdom",
*/