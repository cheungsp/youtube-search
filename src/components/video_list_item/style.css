.single-video-item {
  margin-bottom: 15px;
}