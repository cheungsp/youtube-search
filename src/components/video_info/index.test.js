import React from 'react';
import { shallow } from 'enzyme'
import VideoInfo from './index.js'
import toJson from 'enzyme-to-json'
import './style.css';


test('VideoInfo component should render as expected', () => {
  const video = {
    'etag': "VPWTmrH7dFmi4s1RqrK4tLejnRI/j0uEstXCXOhrDqDegEBmEeHqsBM",
    'snippet': {
      'thumbnails': {
        'default': {
          'url':"https://i.ytimg.com/vi/YQHsXMglC9A/default.jpg"
        }
      }
    },
    'title': 'video 1',
    'key': 1,
    'id': {
      'kind': "youtube#video",
      'videoId': "YQHsXMglC9A"}
  }
  const component = shallow(<VideoInfo video={video}/>)
  const tree =  toJson(component)
  expect(tree).toMatchSnapshot()
})


test('VideoInfo component should render loading... when not passed in any video', () => {
  const component = shallow(<VideoInfo />)
  const tree =  toJson(component)
  expect(component.node.props.children).toBe('Loading...')
})




