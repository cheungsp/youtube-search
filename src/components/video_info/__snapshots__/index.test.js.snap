// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfo component should render as expected 1`] = `
<div
  className="video-detail col-md-8"
>
  <div
    className="embed-responsive embed-responsive-16by9"
  >
    <iframe
      className="embed-responsive-item"
      src="https://www.youtube.com/embed/YQHsXMglC9A"
    />
  </div>
  <div
    className="details"
  >
    <h1 />
    <div
      className="video-info"
    />
  </div>
</div>
`;
